tests/rptest/clients/kubectl.py

```diff
@@ -142,15 +142,34 @@ def logger(self) -> Logger:
     def _local_cmd(self, cmd: list[str]):
         """Run the given command locally and return the stdout as bytes.
            Logs stdout and stderr on failure."""
+        def _prepare_output(sout: str, serr: str) -> str:
+            return f"\n--------- stdout -----------\n{sout}" \
+                   f"\n--------- stderr -----------\n{serr}\n"
+
         self._redpanda.logger.info(cmd)
-        try:
-            return subprocess.check_output(cmd, stderr=subprocess.PIPE)
-        except subprocess.CalledProcessError as e:
-            self.logger.info(
-                f'Command failed (rc={e.returncode}).\n' +
-                f'--------- stdout -----------\n{e.stdout.decode()}' +
-                f'--------- stderr -----------\n{e.stderr.decode()}')
-            raise
+
+        # Using text mode to get strings, not binary
+        # Following code will capture all output to log and work
+        # significantly faster than 'check_output'
+        process = subprocess.Popen(cmd,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   text=True)
+        s_out, s_err = process.communicate()
+        self.logger.debug(_prepare_output(s_out, s_err))
```

Comment:
In the failure case, doesn't this mean we log the output 2x to the debug log?

You could just log once and choose between "debug" and "info" based on whether the process failed or not.

Reply:
Good point. Updated.